airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -193,6 +194,36 @@ def follow_container_logs(self, pod: V1Pod, container_name: str) -> PodLoggingSt
         )
         return self.fetch_container_logs(pod=pod, container_name=container_name, follow=True)
 
+    def log_iterable(self, logs: Iterable[bytes]) -> Optional[DateTime]:
+        for line in logs:
+            timestamp, message = self.parse_log_line(line.decode('utf-8'))
+            self.log.info(message)
+        return timestamp
+
+    def consume_container_logs_stream(self, pod: V1Pod, container_name: str, stream: Iterable[bytes]) -> Optional[DateTime]:
+        async def consume_log_stream() -> Optional[DateTime]:
+            return self.log_iterable(stream)
+
+        async def async_await_container_completion() -> None:
+            while self.container_is_running(pod=pod, container_name=container_name):
+                time.sleep(1)
+
+        await_container_completion = asyncio.create_task(async_await_container_completion())
+        log_stream = asyncio.create_task(consume_log_stream())
+        asyncio.run(asyncio.wait({log_stream, await_container_completion}, return_when=asyncio.FIRST_COMPLETED))
+
+        if log_stream.done():
+            return log_stream.result()
+
+        log_stream.cancel()
+        self.log.warning(
+            "Pod %s log read was interrupted at some point caused by log rotation "
```

Comment:
we can do better here, adding a call to fetch_container_logs(follow=False) to consume the logs we lost after the rotation happened.

However, this will add a bit of an unnecessary overhead: we don't expect that to happen with future versions of k8s that have this api fixed. Also, in that case, if the pod was deleted for some reason, the log will be misleading.

Reply:
Agree.